src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryFileValidator.cs

```diff
@@ -0,0 +1,172 @@
+                }
+
+                if (serverConnection != null)
+                {
+                    bool isLocal = false;
+                    if (string.Compare(GetMachineName(serverConnection.ServerInstance), Environment.MachineName, StringComparison.OrdinalIgnoreCase) == 0)
+                    {
+                        isLocal = true;
+                    }
+
+                    foreach (string filePath in args.FilePaths)
+                    {
+                        bool IsFolder = false;
+                        bool Existing = IsPathExisting(serverConnection, filePath, ref IsFolder);
+
+                        if (Existing)
+                        {
+                            if (IsFolder)
+                            {
+                                errorMessage = string.Format(SR.BackupPathIsFolderError, filePath);
+                                result = false;
+                                break;
+                            }
+                        }
+                        else
+                        {
+                            // If the file path doesn't exist, check if the folder exists
+                            string folderPath = PathWrapper.GetDirectoryName(filePath);
+                            if (isLocal)
+                            {
+                                if (!string.IsNullOrEmpty(folderPath) && !Directory.Exists(folderPath))
+                                {
+                                    errorMessage = string.Format(SR.InvalidBackupPathError, folderPath);
+                                    result = false;
+                                    break;
+                                }
+                            }
+                            else
+                            {
+                                bool isFolderOnRemote = true;
+                                bool existsOnRemote = IsPathExisting(serverConnection, folderPath, ref isFolderOnRemote);
+                                if (!existsOnRemote)
+                                {
+                                    errorMessage = string.Format(SR.InvalidBackupPathError, folderPath);
+                                    result = false;
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    result = false;
+                }
+            }
+            else
+            {
+                result = false;
+            }
+
+            return result;
+        }
+
+        #region private methods
+
+        private static bool IsPathExisting(ServerConnection serverConnection, string path, ref bool isFolder)
+        {
+            Enumerator en = null;
+            DataSet ds = new DataSet();
+            ds.Locale = CultureInfo.InvariantCulture;
+            Request req = new Request();
+            en = new Enumerator();
```

Comment:
>en = new Enumerator(); [](start = 12, length = 22)

why declare this variable as null on line 120 and then init it here?  seems like we should just declare it here where we init?